docs/aep/AEP-2-table-cleanup.md

```diff
@@ -0,0 +1,186 @@
+# AEP-2 Automatic Temporary Table cleanup
+
+
+Authors:
+* Daniel Imberman (daniel@astronomer.io)
+
+
+## Status
+
+| <!-- -->       |  <!-- -->                                                 |
+| -------------- | --------------------------------------------------------- |
+| **State**      | Accepted  |
+| **Discussion** | https://github.com/astronomer/astro-sdk/issues/436                                      |
+| **Created**    | 2022-06-05                                                |
+
+
+## Motivation
+
+When running `aql.transform` or `aql.load_file` jobs, users will develop temporary tables in their databases every time they run a DAG. These extra tables are currently handled by placing all tables into a “temporary schema”. The thought behind these schemas is that a DBA can easily delete the schema if it starts to get full. This answer is insufficient so a better solution for table cleanup should be enacted.
```

Comment:
```suggestion
When running `aql.transform` or `aql.load_file` jobs, the Astro SDK implicitly creates temporary tables in the database every time the DAG runs. These extra tables are currently handled by placing all tables into a “temporary schema”. The thought behind these schemas is that a DBA can easily delete the schema if it starts to get full. This answer is insufficient so a better solution for table cleanup should be enacted.
```

Reply:
@dimberman Could you confirm in the code if we are always storing the tables in the temporary schema, or if there are times when we are actually storing them in the schema set in the connection?
This is the code I believe may be leading to this behaviour: https://github.com/astronomer/astro-sdk/blob/main/src/astro/databases/base.py#L148

Reply:
@tatiana if the user explicitly sets a schema in the connection then yes that will take precedence. This is by design (As we can't assume the same temp schema name across different databases). This also works perfectly with the cleanup step as once we set this up (and soon have a teardown step in the DAG), we won't need temporary schemas anyways.